query/src/main/scala/filodb/query/exec/rangefn/RangeFunction.scala

```diff
@@ -334,13 +372,20 @@ object RangeFunction {
     func match {
       case None                   => () => new LastSampleChunkedFunctionD
       case Some(Last)             => () => new LastSampleChunkedFunctionD
-      case Some(Rate)     if config.fasterRateEnabled => () => new ChunkedRateFunction
-      case Some(Increase) if config.fasterRateEnabled => () => new ChunkedIncreaseFunction
-      case Some(Delta)    if config.fasterRateEnabled => () => new ChunkedDeltaFunction
+      case Some(Increase) if config.fasterRateEnabled && schema.columns(1).isCumulative
+                                  => () => new ChunkedIncreaseFunction
+      case Some(Rate) if config.fasterRateEnabled && schema.columns(1).isCumulative
+                                  => () => new ChunkedRateFunction
+      case Some(Increase) if config.fasterRateEnabled
+                                  => () => new ChunkedPeriodicIncreaseFunction
+      case Some(Rate)     if config.fasterRateEnabled
```

Comment:
Delta types should not need counter correction, so fasterRate does not come into question. We should remove the `if` clause

Reply:
updated the code.